### PR TITLE
doc: add programming matrix table

### PR DESCRIPTION
**Pull request by greg-fer**
Replaced list of supported devices with a matrix.
Fixed MCUBoot > MCUboot typo.

### CODE REVIEW

File: doc/docs/index.md [hunk 1]
```diff
@@ -4,12 +4,27 @@ nRF Connect Programmer is an app available from [nRF Connect for Desktop](https:
...
+| Nordic Thingy:53    | PCA20053   | Yes                      | No                                  | Yes                      |
+| nRF52840 DK         | PCA10056   | Yes                      | No                                  | No                       |
+| nRF52840 Dongle     | PCA10059   | No                       | Yes                                 | No                       |
+| nRF52833 DK         | PCA10100   | Yes                      | No                                  | No                       |
+| nRF52832 DK (nRF52) | PCA10040   | Yes                      | No                                  | No                       |
+| nRF52820 DK         | PCA10115   | No                       | No                                  | No                       |
+| nRF52811 DK         | PCA10068   | No                       | No                                  | No                       |
+| Nordic Thingy:52    | PCA20020   | No                       | No                                  | No                       |
+| nPM1300 EK          | PCA10152   | Yes                      | No                                  | Yes                      |
```
Comment (kylebonnici) [inline doc/docs/index.md:26]: The EK Does nor have a built In Jlink only an external jlink can be used so not sure if this should be true or false

File: doc/docs/index.md [hunk 2]
```diff
@@ -4,12 +4,27 @@ nRF Connect Programmer is an app available from [nRF Connect for Desktop](https:
...
+| nPM1100 EK          | PCA10123   | Yes                      | No                                  | No                       |
```
Comment (kylebonnici) [inline doc/docs/index.md:27]: I never have seen a 1100EK so not sure if it has  Jlink or not ....

File: doc/docs/index.md [hunk 3]
```diff
@@ -4,12 +4,27 @@ nRF Connect Programmer is an app available from [nRF Connect for Desktop](https:
...
+| nRF51 DK            | PCA10028   | No                       | No                                  | No                       |
+| nRF51 Dongle        | PCA10031   | No                       | No                                  | No                       |
+| Custom kits         | n/a        | Yes for all that feature Nordic SoCs  | No                     | No                       |
```
Comment (kylebonnici) [inline doc/docs/index.md:30]: yes for all Nordic SOCs that nrfutil supports so for example nRF52811-QFAAAA is not supported by nrfutil

File: doc/docs/index.md (hunk 1 above)
Comment (greg-fer) [reply doc/docs/index.md:26]: Edited.

**Review by kylebonnici — APPROVED** [review]
Nice :)

**Comment by greg-fer** [comment]
@chunfantasy approved in Teams.

**Pull request merged**